src/main.cpp

```diff
@@ -2567,7 +2567,7 @@ bool DisconnectTip(CValidationState &state, const Consensus::Params &consensusPa
 
     // If this block enabled the may152018 opcodes, then we need to
     // clear the mempool of any transaction using them.
-    if (IsMay152018Enabled(consensusParams, pindexDelete) && !IsMay152018Enabled(consensusParams, pindexDelete->pprev))
+    if (IsMay152018Next(consensusParams, chainActive.Tip()))
```

Comment:
Isnt't these 2 conditions equivalent?

Reply:
you know what , you're right but fact is , it doesn't work...but now i look at it again, i think there is something wrong with pindexDelete->pprev, because if i use it in IsMay152018Next(pindexDelete->pprev) then the script fails.